nautobot/dcim/fields.py

```diff
@@ -56,42 +55,16 @@ def to_python(self, value):
             return value
         if isinstance(value, str):
             value = value.strip()
-        try:
-            return EUI(value, version=48, dialect=mac_unix_expanded_uppercase)
-        except AddrFormatError:
-            raise ValidationError(f"Invalid MAC address format: {value}")
-
-    def get_prep_value(self, value):
-        if not value:
+        if value == "":
```

Comment:
Do we still need the `if value is None:` condition on line 54 if we are already checking that in `get_prep_value()` by `if not value: return ""`?
Also what happens when `value` is "  "? If it satisfies the condition on line 58, that would return `None` is that what we want?

Reply:
The diffs make it a bit confusing here but IIRC this is necessary.

`to_python()` is converting the DB value to a Python value, and this check here is needed to ensure that a DB value of `""` maps to a Python value of `None` (so that the Python value is either an EUI object or None, never a string). I could *probably* rearrange things here to combine the checks from line 54 and line 58 into a single check but I though this was perhaps clearer (at least, when not viewed as a diff)?

Conversely, `get_prep_value()` is converting the Python value to a DB value, thus the change on line 67 below to ensure that the DB value is now `""` rather than `null` since the field is no longer nullable.